pkg/otlp/metrics/metrics_translator.go

```diff
@@ -202,9 +202,17 @@ func (t *Translator) mapNumberMonotonicMetrics(
 			continue
 		}
 
-		if dx, ok := t.prevPts.MonotonicDiff(pointDims, startTs, ts, val); ok {
+		dx, isFirstPoint, shouldDropPoint := t.prevPts.MonotonicDiff(pointDims, startTs, ts, val)
+		if shouldDropPoint {
+			t.logger.Debug("Dropping point: timestamp is older or equal to timestamp of previous point received", zap.String(metricName, pointDims.name))
```

Comment:
Were we logging this before? I think we should avoid logging this, it's in the hot path and the volume of metrics processed by this could be huge. I like the explanation on https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#logging:

> Use logging carefully for events that can happen frequently to avoid flooding the logs. Avoid outputting logs per a received or processed data item since this can amount to very large number of log entries (Collector is designed to process many thousands of spans and metrics per second). For such high-frequency events instead of logging consider adding an internal metric and increment it when the event happens.

Reply:
We weren't logging it before -- Agreed it is likely this will spam the logs, but it will only get logged if it's in `debug` mode. Do you think this is enough of a safeguard, or I should still remove it ?

Reply:
We do have other logs for other unlikely situations below... https://github.com/DataDog/opentelemetry-mapping-go/blame/de6046116470f8e2367b9a38213a7bafe53eee0f/pkg/otlp/metrics/metrics_translator.go#L780-L812 I guess we can keep it and think about switching to metrics/improving this